compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -474,21 +474,24 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
     type TypedArg = Arg
     type Result = Unit
 
+    protected def argOK(arg: TypedArg, formal: Type) = argType(arg, formal) match {
+      case ref: TermRef if ref.denot.isOverloaded =>
+        // in this case we could not resolve overloading because no alternative
+        // matches expected type
+        false
+      case argtpe =>
+        def SAMargOK = formal match {
+          case SAMType(meth) => argtpe <:< meth.info.toFunctionType()
+          case _ => false
+        }
+        isCompatible(argtpe, formal) || ctx.mode.is(Mode.ImplicitsEnabled) && SAMargOK
```

Comment:
Maybe add some note that implicit search and SAM conversion are tried at the same time to justify checking ImplicitsEnabled here?